src/elements/chatbox/chatbox.js

```diff
@@ -0,0 +1,60 @@
+import './chatbox.less'
+import React, { useState, useEffect } from 'react'
+import url from '/util/url'
+
+export const ChatBox = ({ roomId, ...props }) => {
+  const [messageInput, setMessageInput] = useState('')
+  const [chatLog, setChatLog] = useState('')
+  const [chatSocket, setChatSocket] = useState()
+  // const socketRef = useRef()
```

Comment:
I think you were on the right track using a ref for the socket. State should only be used for data that's represented in the JSX/HTML. Any time a state change happens, React will re-run the render to determine if it needs to update the DOM.